src/base/views/viewer.tsx

```diff
@@ -59,19 +60,21 @@ export class ModelRenderer implements RenderingContext {
             vnode);
     }
 
-    renderElement(element: Readonly<SModelElement>, args?: object): VNode | undefined {
+    renderElement(element: Readonly<SModelElement>, args?: IViewArgs): VNode | undefined {
         const view = this.viewRegistry.get(element.type);
-        const vnode = view.render(element, this, args);
+        const vnode = view.render(element, this, {...args, ...this.args});
         if (vnode) {
             return this.decorate(vnode, element);
         } else {
             return undefined;
         }
     }
 
-    renderChildren(element: Readonly<SParentElement>, args?: object): VNode[] {
+    renderChildren(element: Readonly<SParentElement>, args?: IViewArgs): VNode[] {
+        const childArgs = args ? { ...args, parentArgs: this.args } : this.args;
+        const context = new ModelRenderer(this.viewRegistry, this.targetKind, this.postprocessors, childArgs);
         return element.children
-            .map(child => this.renderElement(child, args))
+            .map(child => context.renderElement(child, childArgs))
```

Comment:
Using `childArgs` both as function argument and ModelRenderer property seems redundant and potentially confusing to me. Shall we skip the `args` argument in `renderElement` and just use `this.args` in that method?

Reply:
Agreed, I removed the `childArgs` argument from the `renderElement` call.

Reply:
Shall we also remove the `args` from the `renderElement` signature then?